File: BinaryTree/ts/bottomRightValue.ts
import { TreeNode } from "./TreeNode";

export function bottomRightValue(root: TreeNode<number>): number {
  const queue: TreeNode<number>[] = [root];
  let current: TreeNode<number> = root;

  while (queue.length > 0) {
    current = queue.shift()!;
    if (current.left) queue.push(current.left);
    if (current.right) queue.push(current.right);
  }

  return current.val;
}

File: BinaryTree/go/bottomRightValue.go
package binarytree

// BottomRightValue returns the value of the last node visited in a BFS traversal.
func BottomRightValue(root *TreeNode) int {
	if root == nil {
		return 0
	}

	queue := []*TreeNode{root}
	var current *TreeNode

	for len(queue) > 0 {
		current = queue[0]
		queue = queue[1:]

		if current.Left != nil {
			queue = append(queue, current.Left)
		}
		if current.Right != nil {
			queue = append(queue, current.Right)
		}
	}

	return current.Val
}
